### DISCUSSION TITLE
Is it possible to support installing rdkit (Chemistry Information Package)?

### DISCUSSION THREAD

**Discussion by ShangChien** [pyodide/pyodide#2586] [Q&A]
[rdkit](https://github.com/rdkit/rdkit) package is an important and widely used component of modern chemical molecular drug material design field, which is not a pure python package.
Looking forward to using it directly in the browser.

**Comment by MichelML** [comment]
+1

**Comment by patochinestrad** [comment]
+1

**Comment by ryanking13** [comment]
I think someone who knows about rdkit would need to start building it, then we (Pyodide core devs) can help deal with build issues. AFAIK no Pyodide core devs know rdkit or cheminformatics.

**Reply by hoodmane** [reply]
Getting Cairo to work might by itself be a pretty big deal, though it looks like [someone has done it](https://www.nilsschneider.de/wp/tag/webassembly/).

**Comment by skalkin** [comment]
Initiated a discussion within the RDKit community: https://github.com/rdkit/rdkit/discussions/7827

**Comment by yansenlei** [comment]
Before RDKit supports any `whl`, you can try using the simple [rdkit-js](https://www.rdkitjs.com/) version. Using JS modules with Pyodide can achieve this.

```python

import js

canvas = js.document.createElement('canvas')
js.document.body.append(canvas)

rk = await js.initRDKitModule()
print(rk.version())

smiles = "CC(=O)Oc1ccccc1C(=O)O"
mol = rk.get_mol(smiles)

print('inchi', mol.get_inchi())

smarts = "Oc1[c,n]cccc1"
qmol = rk.get_qmol(smarts)

mdetails = mol.get_substruct_match(qmol)

mol.draw_to_canvas_with_highlights(canvas, mdetails)

mol.get_inchi()
```
